snap/implicit_test.go

```diff
@@ -71,3 +73,174 @@ func (s *InfoSnapYamlTestSuite) TestImplicitSlotsAreRealInterfaces(c *C) {
 		c.Check(known[ifaceName], Equals, true)
 	}
 }
+
+func (s *InfoSnapYamlTestSuite) TestAddImplicitHooksNoHooks(c *C) {
+	dirs.SetRootDir(c.MkDir())
+	defer func() { dirs.SetRootDir("") }()
+
+	yaml := `name: foo
+version: 1.0`
+	info := snaptest.MockSnap(c, yaml, &snap.SideInfo{Revision: snap.R(42)})
+
+	// Now load implicit hooks
+	snap.AddImplicitHooks(info)
```

Comment:
There are other tests that call Read*. I could rewrite these more detailed tests and put them into info_test.go though and make them use Read*, if you prefer (i.e. they probably don't belong in implicit_test.go if they're testing functions in info.go).